test/spec/modules/dfpAdServerVideo_spec.js

```diff
@@ -706,4 +709,167 @@ describe('The DFP video support module', function () {
     expect(customParams).to.have.property('other_key', 'other_value');
     expect(customParams).to.have.property('hb_rand', 'random');
   });
+
+  it('should return unmodified fetched gam vast wrapper if local cache is not used', (done) => {
+    const gamWrapper = (
+      `<VAST version="3.0">` +
+        `<Ad>` +
+          `<Wrapper>` +
+           `<AdSystem>prebid.org wrapper</AdSystem>` +
+            `<VASTAdTagURI><![CDATA[https://endpoint.com]]></VASTAdTagURI>` +
+          `</Wrapper>` +
+       `</Ad>` +
+      `</VAST>`
+    );
+    server.respondWith(gamWrapper);
+
+    getVastXml({})
+      .then((finalGamWrapper) => {
+        expect(finalGamWrapper).to.deep.eql(gamWrapper);
+        done();
+      });
+
+    server.respond();
+  });
+
+  it('should substitue vast ad tag uri in gam wrapper with blob content in data uri format', (done) => {
+    config.setConfig({cache: { useLocal: true }});
+    const url = 'https://pubads.g.doubleclick.net/gampad/ads'
+    const blobContent = '<VAST version="3.0>EXAMPLE VAST BLOB</VAST>';
+    const blobUrl = URL.createObjectURL(new Blob([blobContent], { type: 'text/xml' }));
+    const uuid = generateUUID();
+    const localMap = new Map([[uuid, blobUrl]]);
+
+    const bidCacheUrl = 'https://prebid-test-cache-server.org/cache?uuid=' + uuid;
+    const gamWrapper = (
+      `<VAST version="3.0">` +
+        `<Ad>` +
+          `<Wrapper>` +
+           `<AdSystem>prebid.org wrapper</AdSystem>` +
+            `<VASTAdTagURI><![CDATA[${bidCacheUrl}]]></VASTAdTagURI>` +
+          `</Wrapper>` +
+       `</Ad>` +
+      `</VAST>`
+    );
+
+    const dataUrl = `data://text/xml;base64,${btoa(blobContent)}`;
+    const expectedOutput = (
+      `<VAST version="3.0">` +
+        `<Ad>` +
+         `<Wrapper>` +
+            `<AdSystem>prebid.org wrapper</AdSystem>` +
+            `<VASTAdTagURI><![CDATA[${dataUrl}]]></VASTAdTagURI>` +
+          `</Wrapper>` +
+       `</Ad>` +
+      `</VAST>`
+    );
+
+    server.respondWith(/^https:\/\/pubads.*/, gamWrapper);
+    server.respondWith(/^blob:http:*/, blobContent);
+
+    getVastXml({url, adUnit: {}, bid: {}}, localMap)
+      .then((vastXml) => {
+        expect(vastXml).to.deep.eql(expectedOutput);
+        done();
+      })
+      .catch(done)
```

Comment:
why is this line necessary ? Wouldn't we want the test to fail if an exception is thrown ?